game-app/game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/OffensiveSubsRetreat.java

```diff
@@ -34,17 +34,12 @@
 
   @Override
   public List<String> getNames() {
+    // do not check for destroyer because it could die before the retreat step
```

Comment:
This comment has me a bit confused. Is it here because of historical context, or is it going to be obvious in some other way that there *should* be a check for a destroyer here that would otherwise make a maintainer want to insert one? Perhaps the comment can be removed? It does seem like it makes sense if you knew what code used to exist here before, but maybe not if you look at the very latest.

Reply:
The comment is needed.  In most of the battle steps, the conditions for the "getNames" and for "execute" are almost always identical.  But for this step, that isn't the case.  So the comment explains why the destroyer isn't checked in "getNames" even though it will be checked in "execute".

Reply:
Cool, perhaps a quick prefix to this comment would clarify that, eg:
"In most battle steps we check for destroyers, but here we do not because destroyers may die before the retreat step".

I think a logical extension though is "so what if the destroyers died before?". A second sentence on the comment to explain the implication would make that very clear. Let me know if that ask does not make sense.